src/borg/locking.py

```diff
@@ -158,6 +161,41 @@ def is_locked(self):
     def by_me(self):
         return os.path.exists(self.unique_name)
 
+    def kill_dead_lock(self):
+        for name in os.listdir(self.path):
+
+            try:
+                host_pid, thread = name.rsplit('-', 1)
+                host, pid = host_pid.rsplit('.', 1)
+            except ValueError:
+                # Malformed lock name? Or just soem new format we don't understand?
+                # It's safer to just exit
+                return False
+
+            if host != _hostname:
+                return False
+
+            try:
+                # This may not work in Windows.
+                os.kill(int(pid), 0)
```

Comment:
Ha, no, it does not kill anything. it sends a signal, and signal 0 has a special meaning of "don't send anything, just do all the checks" and is frequently used as a way to see if a particular process is there or not. Certainyl beats spawning "ps -ef | grep pid" and other such sily techniques that are also a lot less portable across OSes.

I will add a comment to this effect.